### ISSUE TITLE
Demo: add new art assets 

### ISSUE THREAD

**Issue by m7600** [gemrb/gemrb#1078]
This issue is for adding art assets to the demo. I think it's more convenient to post them here before making a pull request for them, because they may need to be changed or fixed in one way or another. 

My plan is to make animations for the following:
Human
Elf
Dwarf
Halfling
Gnome
Half-orc

Each of them needs a male and a female version. All of these would be basic (which means no weapons and no metal armor, for example, those things should be added later. For now, just basic clothes).
Half-Elves are not on the list because, if I remember correctly, in the Infinity Engine games they use the same animations as the Elves, but I could be wrong. The same applies to variations such as Drow, Duergar, etc.
One thing I'm not entire sure about is Tieflings. I don't remember what Haer Dalis's animation looks like in BG2, but he probably uses the same one as Humans or Elves. In IWD2, I think they use the same one as Humans. An exception to this is PS:T, since Annah has a demon-tail. Not sure which way we should go with Tieflings.

**Comment by lynxlynxlynx** [comment]
For starters and the demo, one good animation for the pc is more than enough. Just the standing and walking cycles would do for now. Idle (bored) as a bonus.

**Comment by m7600** [comment]
Technical stuff:

Looking at the animations for the protagonist, there are 18 .bam files. The new assets will have to have the same number of files, each. These are:

A1: Attack 1. Single handed, diagonal downward chop.
A2: Attack 2. Single handed, straight downward chop.
A3: Attack 3. Single handed stab.
A4: Attack 4. Two-handed, diagonal downward chop.
A5: Attack 5. Two-handed, straight downward chop.
A6: Attack 6. Two-handed stab.
A7: Attack 7. This one is weird, the character seems to fall down.
A8: Attack 8. Single handed, straight open palm strike.
CA: Cast a spell.
DE: Death.
GH: Get hit.
GU: Get up.
SC: Stance, ready for combat.
SD: Standing, idle.
SL: Sleep.
SP: This one is weird. It could be a spell or a "special". 
TW: Tweak, or twitch on the ground.
WK: Walk.

This means that, considering the types of characters listed before, male and female, we need a total of 216 .bam files for basic character animations.

**Comment by m7600** [comment]
@lynxlynxlynx ok, good to know. I'll still make the other ones as well. Maybe we can start to think about a new game, different from the demo (no rush, of course, all of this is going to take time)

**Comment by lynxlynxlynx** [comment]
One thing at a time or nothing will get done. Incremental steps will get us to a fully fledged demo. Once we're there, which is no small feat, only then it makes sense to set our sights on a whole new game, a much much more daunting task.

I suggest you focus on making the human animation perfect first, which will also help you fine-tune and document the workflow. The better we understand it, the easier it is to streamline. Then we choose the next task and repeat. 

I guess it would help if we drafted a design for the demo, which can then be a base for a roadmap and to know when it's done. I'll continue this thought in the other issue.

btw: 
- DE death: IIRC this is the dying animation, so falling to the ground. TW is the dead one, lying on the ground.
- main attack animation stances are for slashing, backslashing and jabbing; plus shooting (ranged)
- hmm, iwd didn't have head turning?

**Comment by chilvence** [comment]
Yeah, I'd also just focus on getting one character looking really good. I'd really recommend spending a lot of time on the movement cycles. Animation is probably more than half of what makes a character look real. Your model can be a stick figure with a sphere head but still look convincing if it is nicely animated. Besides that, once you are happy, you can just re-use the skeletal rig on the other species.

One tip I suggest, add a key frame pose before and after the actual set of frames to render ; without it, the start and end frames of a cycle won't interpolate well with each other, because they will be blending back to a straight pose.

Also, Disney has a dirty secret making natural looking animated characters, they used rotoscoping liberally, ie, filming an actor and then tracing over the result. The technique worked very well when Jordan Mechner used it for the original Prince of Persia, at the time it blew everyone away by how realistic it looked :) (and that was a three colour sprite with slightly more pixels than a mouse cursor)

**Comment by lynxlynxlynx** [comment]
@m7600 Hey, have you made any progress on the new base animation?

**Comment by lynxlynxlynx** [comment]
Found some very nice animations on OGA, with several stances. Could be useful for enemies once we get that far:
https://opengameart.org/users/marcintokarski

**Comment by m7600** [comment]
Sorry @lynxlynxlynx, I've been swamped with work from my day job lately, I'll get to the animations eventually. But here are some buildings that I made several months ago, I uploaded them to opengameart.org as CC0 (Public Domain). Maybe they can be used for a new area in the future.

[](https://opengameart.org/content/mali-buildings)

**Comment by chilvence** [comment]
Those buildings look bloody brilliant, I definitely feel a strong Age of Empires vibe coming from them :)

Anyway, don't put your yourself on the hook, everyone has to survive 👍

**Comment by lynxlynxlynx** [comment]
Well 0 A.D. is basically AoE on steroids. :)

I've made a lot of progress on the first area btw (#313), which also made it more obvious what needs to be done overall. So unfortunately, as often as I tick checkboxes as done, I'm adding new ones to the later stages. I'll be replacing the main GUI elements with the ones from the RPG GUI pack (what we already use in the map window), but more GUI work will wait for subviews, since otherwise there's just a bit too much work with cleaning things up after the merge.

OGA has been great for resources. I couldn't find a tiny swarm animation or anything else that would add flair as a static area animation near the hidden cache, we don't support enough colors nor PNG spritesheets, but other than that, it's been a blast. We actually have gameplay now.

If either of you could export that rabbit or rat blender models, that would be great — two more ticks less. They're already rigged and animated.

**Comment by m7600** [comment]
@lynxlynxlynx here's a zip file with 8 .bam files. They are: humans, elves, dwarves, and orcs, each of them has a female and male versions, and two types of actions: standing and walking. Maybe these could be used for NPCs for Stage X. The male orc animations have some problems, I'll have to edit it later. All of them are wearing the same basic leather armor.
[char_animations.zip](https://github.com/gemrb/gemrb/files/6194346/char_animations.zip)

**Comment by lynxlynxlynx** [comment]
You're very prolific. 👍🏾 We'll see when we get there — can't say much more, as I have no clue either. 🤷🏾
